### PR TITLE
#47 enable propTypes setter on returned component

### PR DESCRIPTION
**Pull request by dmaj7no5th**
The React component returned by reactify doesn't support setting of propTypes, which breaks the expected React API. Trying to set propTypes on the Component it returns throws a runtime error:

```
const ReactComponent = reactify(MyWebComponent);

ReactComponent.propTypes = {
  foo: React.PropTypes.string
};

//=> TypeError: Cannot set property propTypes of function ReactComponent() {
		      _classCallCheck(this, ReactComponent);
		
		      return _possibleConstructorR...<omitted>... } which has only a getter
```

This changes just adds a setter method that uses Object.assign under the hood to apply propTypes passed in to the default propTypes that were being defined already.

### CODE REVIEW

**Comment by jpnelson** [comment]
lgtm!

**Comment by dmaj7no5th** [comment]
Changed PR to just remove the default propTypes that were being set. They weren't really adding any value. This way there's no confusion on what's being set as per @treshugart's comments on the issue.

File: test/unit/prop-types.js [hunk 1]
```diff
@@ -3,8 +3,12 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 
 describe('prop-types', () => {
-  it('children', () => {
+
```
Comment (stevemao) [inline test/unit/prop-types.js]: Can you remove unnecessary whitespace change?

**Comment by treshugart** [comment]
I wonder if `git cz --amend` works?

**Comment by stevemao** [comment]
Otherwise looks good 😄

**Comment by dmaj7no5th** [comment]
@treshugart is it really a breaking change though?

**Comment by treshugart** [comment]
@dmaj7no5th I think, technically yes. If someone is relying on those props to be set and now they have to manually define them, that'd be breaking, right?

File: test/unit/prop-types.js (hunk 1 above)
Comment (dmaj7no5th) [inline test/unit/prop-types.js]: fixed

**Comment by dmaj7no5th** [comment]
All cleaned up. Lemme know if the commit messaging is okay

**Comment by treshugart** [comment]
Seems good but I just noticed in https://github.com/skatejs/named-slots/commits/master that `BREAKING CHANGE` alone didn't seem to trigger it. I think it needs a colon after it, so `BREAKING CHANGE:`. They do that here https://github.com/semantic-release/semantic-release#major-breaking-release.

Honestly, though, I think I'm just going to merge it and bump manually. I don't have time to get semantic-release working right now. Thanks for the PR @dmaj7no5th!
